---
title: "Lab2"
author: "Katherine Chacon-Vargas"
date: "1/29/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
#Excersise 
#E1

vec.1 <- c(1,3,6,9,12)
vec.2 <- c(1,0,1,0,1)

vec.sum <- vec.1 + vec.2
vec.sum
vec.subs <- vec.1 - vec.2
vec.subs
vec.mult <- vec.1 * vec.2
vec.mult
vec.div <- vec.1 / vec.2
vec.div

#E2
vec.3 <- c(0,1,2,3)
str(vec.3) #num
vec.4 <- c("aa","bb","cc","dd")
str(vec.4) #chr
vec.5 <- c("aa",1,"bb",2)
str(vec.5) #chr

#E3
genotype1 <- c("AA","AA","AG","GG","GG")
genotype2 <- c("AA", "AA", "GG", "GG", "GG")
genotype <- cbind(genotype1,genotype2)
table(genotype) 
#genotype
#AA AG GG 
#4  1  5

#E4
time <- c(0,2,4,6,8)
treatment1 <- c(0,1,2,3,4)
treatment2 <- c(0,2,4,6,8)
treatment3 <- c(0,3,6,9,12)
treatment <- as.data.frame(cbind(treatment1,treatment2,treatment3))
treatment$Time <- c(0,2,4,6,8)
treatment
plot(treatment$Time,treatment$treatment3, xlab = "Time", ylab = "Treatment 3")

#E5
SNP_23me <- read.table("23andMe_complete.txt", header =  T, sep = "\t")
str(SNP_23me)
str(SNP_23me$chromosome) # Factor 
#chromosome object type in truncated file -> int only one level
#chromosome object type in big file -> Factor has multiple levels 

#E6
SNP_23me_genotype <- table(SNP_23me$genotype)
SNP_23me_genotype

#E7
SNP_23me_genotypeA <- subset(SNP_23me, genotype =='A')
head(SNP_23me_genotypeA, n=10)
str(SNP_23me_genotypeA)
table(SNP_23me_genotypeA$chromosome) #chromosome X, Y, MT are haploid with only one copy of the genome, MT has 732 'A', X has 5782 'A' & Y has 162 'A'
GenotypeA_table <- table(SNP_23me_genotypeA$chromosome)

#Table for Frequency of genotype A
GenotypeA_table <- as.data.frame(GenotypeA_table)
GenotypeA_table <- GenotypeA_table[-c(1:22), ]
names(GenotypeA_table)[1] <- paste("CHR")
GenotypeA_table
```


```{r pressure, echo=FALSE}




```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

